.github/workflows/build-and-upload.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Upload Artifacts
+
+on:
+  workflow_call:
+    secrets:
+      npm_token:
+        required: true
+
+jobs:
+  build-and-upload:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Run Setup
+        uses: ./.github/actions/setup
+        with:
+          npm_token: ${{ secrets.npm_token }}
+      - name: Build
+        shell: bash
+        run: yarn lerna run build --since=origin/master --include-dependencies
```

Comment:
this is relevant only for branches that aren't master, maybe we can emphasis it somehow so no one will use it for master?

Reply:
yep :)
#2006 does exactly that and I will update all of our workflows once it is merged, including this new build-and-upload workflow